sdktests/server_side_events_users.go

```diff
@@ -153,58 +153,57 @@ func doServerSideEventUserTests(t *ldtest.T) {
 	}
 }
 
-func eventUserMatcher(user lduser.User, eventsConfig servicedef.SDKConfigEventParams) m.Matcher {
-	// This simulates the expected behavior of SDK event processors with regard to redacting
-	// private attributes. For more details about how this works, please see the SDK
-	// documentation, and/or the implementations of the equivalent logic in the SDKs
-	// (such as https://github.com/launchdarkly/go-sdk-events).
-
-	// First, get the regular JSON representation of the user, since it's simplest to treat
-	// this as a transformation of one JSON object to another.
-	allJSON := ldvalue.Parse(jsonhelpers.ToJSON(user))
-	allAttributes := append(allJSON.Keys(), allJSON.GetByKey("custom").Keys()...)
-
-	expected := []m.KeyValueMatcher{
-		m.KV("kind", m.Equal("user")),
+// eventUserMatcher returns a Matcher to verify that a JSON object has the expected properties based on
+// the input context and the events configuration. This mostly means verifying that private attributes
+// behave correctly.
+//
+// Because the rules for private attributes are fairly complicated, we have not reimplemented that logic
+// here. Instead, this function depends on the Go SDK's implementation of event context formatting,
+// treating it as a reference implementation that has been validated by its own thorough tests.
```

Comment:
This is an oversimplified implementation that only works for private attributes that are specified by a simple attribute name; it doesn't work for attribute reference paths. I think probably I'll end up getting rid of this function and just directly declaring what the expected context JSON is as part of the test parameters, but for now, this is the simplest change to let the current test logic work.